pkg/deploy/ingress.go

```diff
@@ -28,11 +30,18 @@ func NewIngress(cr *orgv1.CheCluster, name string, serviceName string, port int)
 	ingressDomain := cr.Spec.K8s.IngressDomain
 	ingressClass := util.GetValue(cr.Spec.K8s.IngressClass, DefaultIngressClass)
 	labels := GetLabels(cr, name)
-	tlsSecretName := cr.Spec.K8s.TlsSecretName
+
 	tls := "false"
 	if tlsSupport {
 		tls = "true"
 	}
+	tlsSecretName := cr.Spec.K8s.TlsSecretName
+	// If TLS is turned on but the secret is not set, try to use Che default value as k8s cluster defaults will not work.
+	tlsFlag, _ := strconv.ParseBool(tls)
+	if tlsFlag && tlsSecretName == "" {
```

Comment:
It can also use util.GetValue(stringFromCR, defaultStringValue)

 Than code could be:
```
tlsSecretName := cr.Spec.K8s.TlsSecretName
tlsFlag, _ := strconv.ParseBool(tls)
if tlsFlag {
        tlsSecretName = util.getValue(tlsSecretName, "che-tls")
}
```

But it optional, up to you.